app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx

```diff
@@ -122,11 +133,25 @@ class NewsfeedPage extends React.Component<Props, State> {
     localAPI.get(`/news?coinSlugs=${coinSlug}`).then((response) => {
       this.setState({
         status: STATUSES.READY,
-        newsItems: response.payload,
+        sortedNewsItems: response.payload.sort(this.sortNewsFunc),
       })
     })
   }
 
+  fetchMoreNewsItems = () => {
+    const lastNews = this.state.sortedNewsItems[this.state.sortedNewsItems.length - 1];
+    this.setState({
+      status: STATUSES.INFINITE_SCROLL_LOADING
+    });
+
+    localAPI.get(`/news`, { publishedUntil: lastNews.publishedUntil }).then(response => {
+        this.setState({
+          status: STATUSES.READY,
+          sortedNewsItems: [...this.state.sortedNewsItems, ...response.payload.sort(this.sortNewsFunc)]
+        })
+    });
```

Comment:
You should trigger this from within callback of `setState`.

Reply:
Done.